default.nix

```diff
@@ -228,113 +183,69 @@ let overrideCabal = pkg: f: if pkg == null then null else haskellLib.overrideCab
-        reflex-dom = addReflexOptimizerFlag (doJailbreak reflexDom.reflex-dom);
-        reflex-dom-core = addReflexOptimizerFlag (doJailbreak reflexDom.reflex-dom-core);
         reflex-todomvc = self.callPackage (hackGet ./reflex-todomvc) {};
         reflex-aeson-orphans = self.callPackage (hackGet ./reflex-aeson-orphans) {};
-        haven = self.callHackage "haven" "0.2.0.0" {};
-
-        inherit (jsaddlePkgs) jsaddle jsaddle-clib jsaddle-wkwebview jsaddle-webkit2gtk jsaddle-webkitgtk;
-        jsaddle-warp = dontCheck jsaddlePkgs.jsaddle-warp;
 
-        jsaddle-dom = overrideCabal (self.callPackage (hackGet ./jsaddle-dom) {}) (drv: {
-          # On macOS, the jsaddle-dom build will run out of file handles the first time it runs
-          preBuild = ''./setup build || true'';
-        });
+        # Broken Haddock - Please fix!
+        # : error is: haddock: internal error: internal: extractDecl
+        # No idea where it hits?
+        reflex-dom = dontHaddock (addReflexOptimizerFlag reflexDom.reflex-dom);
+        reflex-dom-core = dontHaddock (addReflexOptimizerFlag reflexDom.reflex-dom-core);
 
-        inherit (ghcjsDom) ghcjs-dom-jsffi;
+        inherit (jsaddlePkgs) jsaddle jsaddle-clib
+                              jsaddle-webkit2gtk jsaddle-webkitgtk
+                              jsaddle-wkwebview;
 
-        # TODO: Fix this in Cabal
-        # When building a package with no haskell files, cabal haddock shouldn't fail
-        ghcjs-dom-jsaddle = dontHaddock ghcjsDom.ghcjs-dom-jsaddle;
-        ghcjs-dom = dontHaddock ghcjsDom.ghcjs-dom;
-
-        inherit (gargoylePkgs) gargoyle gargoyle-postgresql;
+        # another broken test
+        # phantomjs has issues with finding the right port
+        # jsaddle-warp = dontCheck (addTestToolDepend jsaddlePkgs.jsaddle-warp nixpkgs.phantomjs);
+        jsaddle-warp = dontCheck jsaddlePkgs.jsaddle-warp;
 
-        ########################################################################
-        # Tweaks
-        ########################################################################
-        gi-glib = self.callPackage ./gi-glib.nix {};
-        gi-gio = self.callPackage ./gi-gio.nix {};
-        gi-gtk = self.callPackage ./gi-gtk.nix {
-          gtk3 = nixpkgs.gnome3.gtk;
-        };
-        gi-javascriptcore = self.callPackage ./gi-javascriptcore.nix {};
-        gi-webkit2 = self.callPackage ./gi-webkit2.nix {
-          webkitgtk = nixpkgs.webkitgtk216x;
-        };
-        gi-gtksource = super.gi-gtksource.override {
-          inherit (nixpkgs.gnome3) gtksourceview;
-        };
-        ghcjs-base-stub = dontHaddock super.ghcjs-base-stub;
+        jsaddle-dom = self.callPackage (hackGet ./jsaddle-dom) {};
 
-        haskell-gi-overloading = super.haskell-gi-overloading_0_0;
+        haskell-gi-overloading = dontHaddock (self.callHackage "haskell-gi-overloading" "0.0" {});
```

Comment:
Looks like this is in fact redundant, @ryantrinkle. `0.0` is the version that nixpkgs has by default, it seems. Not sure if it's a good idea to rely on nixpkgs to be right about this though; so pinning it manually may be a good idea nonetheless. Not sure

Reply:
I'd prefer to keep it here, or alternatively we could *assert* it here.  The failure mode is that everything gets *really* slow to build, which is not obviously linked to this.